### PR TITLE
feat: replace openjdk with amazoncorretto:17.0.4 for security compliance

### PR DESCRIPTION
**Pull request by cuyk**
## What
We want to improve the security rating of our docker images.

## How
Switch to amazoncorretto:17.0.4 docker image as base one. It was created with Amazon Linux with all the latest patches.

### CODE REVIEW

**Review by git-phu — COMMENTED** [review]
looking good!

Two questions:
1. do we have a comparison of what open vulnerabilities are present in `amazoncorretto:17.0.4` vs `openjdk:19-slim-bullseye`?
2. We'll need to update the main `build.gradle` file as well [here](https://github.com/airbytehq/airbyte/blob/master/build.gradle#L232-L242).  (This is where we actually pass the build args to Docker).

File: buildSrc/src/main/groovy/airbyte-docker.gradle [hunk 1]
```diff
@@ -117,8 +117,8 @@ class AirbyteDockerPlugin implements Plugin<Project> {
         return "$stdout".toString().trim()
     }
 
-    // Some image tags rely on environment variables (e.g. "FROM openjdk:${JDK_VERSION}-slim").
-    // dump those into a "sh -c 'echo ...'" command to resolve them (e.g. "openjdk:17-slim")
+    // Some image tags rely on environment variables (e.g. "FROM eclipse-temurin:${JDK_VERSION}-jdk-jammy").
+    // dump those into a "sh -c 'echo ...'" command to resolve them (e.g. "eclipse-temurin:17-jdk-jammy")
```
Comment (supertopher) [inline buildSrc/src/main/groovy/airbyte-docker.gradle:121]: At standup it sounded to me like we were abandoning `eclipse` for `corretto` is the use of `eclipse` intentional here?

File: tools/bin/build_image.sh [hunk 2]
```diff
@@ -42,7 +42,7 @@ if [ "$FOLLOW_SYMLINKS" == "true" ]; then
   # to use as the build context
   tar cL "${exclusions[@]}" . | docker build - "${args[@]}"
 else
-  JDK_VERSION="${JDK_VERSION:-17.0.1}"
+  JDK_VERSION="${JDK_VERSION:-17}"
```
Comment (supertopher) [inline tools/bin/build_image.sh:45]: I have no idea if we care about the patch version in practice, but I'd like to stick with what we have already unless we have a good reason to move.  I also see we are using `17.0.4` elsewhere.  

I don't understand fully how these are related to each other in practice though

File: airbyte-workers/Dockerfile [hunk 3]
```diff
@@ -1,31 +1,15 @@
...
-# See https://github.com/airbytehq/airbyte/issues/8643 for more information on why we are using kubectl for copying.
-# The following commands were taken from https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-using-native-package-management
-RUN wget -O /usr/share/keyrings/kubernetes-archive-keyring.gpg https://packages.cloud.google.com/apt/doc/apt-key.gpg && \
-        echo "deb [signed-by=/usr/share/keyrings/kubernetes-archive-keyring.gpg] https://apt.kubernetes.io/ kubernetes-xenial main" | tee /etc/apt/sources.list.d/kubernetes.list
-RUN apt-get update && apt-get install -y kubectl
+RUN amazon-linux-extras install -y docker
+RUN yum install -y jq tar
+RUN yum clean all
+
+RUN curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/${DOCKER_BUILD_ARCH}/kubectl" \
```
Comment (supertopher) [inline airbyte-workers/Dockerfile:11]: curlception.

**Review by supertopher — COMMENTED** [review]
When we roll this out it would be good to test in limited to `dev` instance only.  I don't know how we practically accomplish this though.

nothing major in here, I just want to make sure we don't throw a new base image into production without more robust testing.  I also have peter's questions around what the security report looks like for this VS ubuntu

**Comment by supertopher** [comment]
when you want another review you may have to ping us on slack.  I don't think GitHub will ping us on slack and we are drowning in GH notifications

**Comment by cuyk** [comment]
>     1. do we have a comparison of what open vulnerabilities are present in `amazoncorretto:17.0.4` vs `openjdk:19-slim-bullseye`?
Snyk showed no low/medium/critical issues on `amazoncorretto:17.0.4` at all. There are 40 issues on local snyk run with `openjdk:19-slim-bullseye`

>     2. We'll need to update the main `build.gradle` file as well [here](https://github.com/airbytehq/airbyte/blob/master/build.gradle#L232-L242).  (This is where we actually pass the build args to Docker).
Updated. Thanks for the hint!

File: buildSrc/src/main/groovy/airbyte-docker.gradle (hunk 1 above)
Comment (cuyk) [reply buildSrc/src/main/groovy/airbyte-docker.gradle]: Missed it switching to the initial plan with `amazoncorretto`. Fixxed

**Review by supertopher — COMMENTED** [review]
this looks good to me, I want to leave approval to Davin or Peter though.  I don't have enough context on how these are consumed downstream

**Comment by davinchia** [comment]
@cuyk before we approve, can we remove all changes in the integrations folder? 

If we want to make sure the change works, we would have to test all the integrations this changes affects, which is alot!

**Comment by davinchia** [comment]
@cuyk did we replace this with another PR?

**Pull request closed without merging**